include/mbedtls/hkdf_tls.h

```diff
@@ -0,0 +1,162 @@
+/**
+* \file hkdf_tls.h
+*
+* \brief TLS 1.3-specific HKDF functionality
+*
+*  Copyright (C) 2006-2015, ARM Limited, All Rights Reserved
+*  SPDX-License-Identifier: Apache-2.0
+*
+*  Licensed under the Apache License, Version 2.0 (the "License"); you may
+*  not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*  http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+*  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+*  This file is part of mbed TLS (https://tls.mbed.org)
+*/
+#ifndef MBEDTLS_HKDF_TLS_H
+#define MBEDTLS_HKDF_TLS_H
+
+#if !defined(MBEDTLS_CONFIG_FILE)
+#include "mbedtls/config.h"
+#else
+#include MBEDTLS_CONFIG_FILE
+#endif
+
+//#include "mbedtls/md.h"
+#include "mbedtls/ssl.h"
+#include "mbedtls/ssl_internal.h"
+
+//#include "sha_new.h"
+
+#define MBEDTLS_ERR_HKDF_BUFFER_TOO_SMALL                  -0x6A00  /**< A buffer is too small to store label */
+#define MBEDTLS_ERR_HKDF_BAD_INPUT_DATA                    -0x7100  /**< Bad input parameters to function.  */
+#define MBEDTLS_ERR_HKDF_ALLOC_FAILED                      -0x4D80  /**< Memory allocation failed. */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+
+/**
+ * \brief           hkdfEncodeLabel creates the HkdfLabel structure.
+ *
+ * \param label     Label Text
+ * \param llen      Label Length
+ * \param hashValue Hash Value
+ * \param hlen      Length of hash value
+ * \param buf       Output buffer
+ * \param length    Length is a value encoded in the HkdfLabel structure.
+ *
+ * \return          0 if successful,
+ *                  or MBEDTLS_ERR_HKDF_BUFFER_TOO_SMALL
+ *
+ * \note            TLS 1.3 encodes the labels for the HKDF function
+ *                  in a unique way. This function allows to
+ *                  conveniently create these labels.
+ *
+ *                  The function assumes that enough buffer has been
+ *                  allocated in buf to hold the result.
+ */
+int hkdfEncodeLabel(const unsigned char *label, int llen,
```

Comment:
None of the functions in this header follow the mbed TLS naming style or conventions!